src/endpoint/s3/ops/s3_put_bucket_lifecycle.js

```diff
@@ -94,16 +94,14 @@ async function put_bucket_lifecycle(req) {
         }
         current_rule.status = rule.Status[0];
 
-        if (rule.Prefix) {
-            dbg.error('Rule should not have prefix, it should be filter.prefix', rule);
-            throw new S3Error(S3Error.InvalidArgument);
-        }
-
-        if (!(rule.Filter && rule.Filter.length === 1)) {
-            dbg.error('Rule should have filter', rule);
+        if (rule.Prefix && rule.Prefix.length === 1) {
+            current_rule.filter = { prefix: rule.Prefix[0] };
+        } else if (rule.Filter && rule.Filter.length === 1) {
+            current_rule.filter = parse_filter(rule.Filter[0]);
```

Comment:
Why do you use `parse_filter` in line 100 but not in line 98?

Reply:
according to aws documentation there can be either prefix or filter, but not both. (prefix field is deprecated but still available). so no need to parse_filter. I can add a check that checks we don't have both

Reply:
Please add a comment on this line: `current_rule.filter = { prefix: rule.Prefix[0] }`